crates/ruff_python_ast/src/parenthesize.rs

```diff
@@ -0,0 +1,42 @@
+use std::ops::Sub;
+
+use ruff_python_trivia::{SimpleTokenKind, SimpleTokenizer};
+use ruff_text_size::{TextRange, TextSize};
+
+use crate::node::AnyNodeRef;
+use crate::Ranged;
+
+/// Returns the [`TextRange`] of a given expression including parentheses, if the expression is
+/// parenthesized; or `None`, if the expression is not parenthesized.
+pub fn parenthesized_range(
```

Comment:
I got rid of the struct in favor of a simple function that returns `Option`. I've gone back and forth on it a little bit.